pkg/controller/migration/convert/core.go

```diff
@@ -255,20 +257,36 @@ func handleAnnotations(c *components, _ *operatorv1.Installation) error {
 
 // removeExpectedAnnotations returns the given annotations with common k8s-native annotations removed.
 // this function also accepts a second argument of additional annotations to remove.
-func removeExpectedAnnotations(existing, ignore map[string]string) map[string]string {
+func removeExpectedAnnotations(existing, ignoreWithValue map[string]string, ignoreNoMatterWhatValue []string) map[string]string {
 	a := existing
+	annotationKeyRegexps := make([]*regexp.Regexp, 0)
+	for _, annotationKey := range ignoreNoMatterWhatValue {
+		annotationKeyRegexp, err := regexp.Compile(annotationKey)
+		if err != nil {
+			log.Error(fmt.Errorf("%s is not a valid regular expression", annotationKey), "Error when removing expected annotations")
+			continue
+		}
+		annotationKeyRegexps = append(annotationKeyRegexps, annotationKeyRegexp)
+	}
+
 	for key, val := range existing {
-		if key == "kubectl.kubernetes.io/last-applied-configuration" ||
-			key == "deprecated.daemonset.template.generation" ||
+		if key == "deprecated.daemonset.template.generation" ||
 			key == "deployment.kubernetes.io/revision" ||
 			key == "scheduler.alpha.kubernetes.io/critical-pod" {
 			delete(a, key)
 			continue
 		}
 
-		if v, ok := ignore[key]; ok && v == val {
+		if v, ok := ignoreWithValue[key]; ok && v == val {
 			delete(a, key)
 		}
```

Comment:
Seems like we could add a `continue` here as well, similar to the first if-block, since there's a new 3rd block that we can skip (if we have entered this second if-block).